app/HutchAgent/Services/WorkflowTriggerService.cs

```diff
@@ -89,7 +88,41 @@ private WfexsJob UnpackCrate(Stream stream)
       if (File.Exists(mainEntityPath) && (mainEntityPath.EndsWith(".stage") || mainEntityPath.EndsWith(".yaml") ||
                                           mainEntityPath.EndsWith(".yml")))
       {
+        _logger.LogInformation($"main Entity is a Wfexs stage file and can be found at {mainEntityPath}");
         _workflowOptions.StageFilePath = mainEntityPath;
+        // Create a copy of the wfexs stage file
+        var copyFilePath = Path.Combine(wfexsJob.UnpackedPath, "copy_" + mainEntity.Id);
+        try
+        {
+          File.Copy(mainEntityPath,copyFilePath, true);
+        }
+        // Catch exception if the file was already copied.
+        catch (IOException copyError)
+        {
+          _logger.LogError(copyError.Message);
```

Comment:
Just to check, are you happy to continue with the method if the file was already copied - is an error log sufficient or should this end process as well?

Reply:
I'm not sure, since technically this would be copying it into a clean directory in every case now, which gets created at the start of the method. Maybe there's no point in logging this error and instead just end the process if the copying fails